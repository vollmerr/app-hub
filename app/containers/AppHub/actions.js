/*
 *
 * AppHub actions
 *
 */

import {
  CHANGE_MOBILE,
} from './constants';

export function changeMobile(isMobile) {
  return {
    isMobile,
    type: CHANGE_MOBILE,
  };
}
