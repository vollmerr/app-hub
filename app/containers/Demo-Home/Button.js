import React from 'react';

const Button = ({ onClick, disabled, text }) => ( // eslint-disable-line
  <button
    onClick={onClick}
    disabled={disabled}
    style={{ background: disabled ? '#666' : '#333', color: '#fff', padding: '10px 15px', cursor: 'pointer' }}
  >{text}</button>
);

export default Button;
