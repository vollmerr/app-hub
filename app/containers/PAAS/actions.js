import { createAction } from 'redux-actions';

import * as C from './constants';


export const getManagerDataRequest = createAction(C.GET_MANAGER_DATA_REQUEST);
export const getManagerDataSuccess = createAction(C.GET_MANAGER_DATA_SUCCESS);
export const getManagerDataFailure = createAction(C.GET_MANAGER_DATA_FAILURE);

export const getReportDataRequest = createAction(C.GET_REPORT_DATA_REQUEST);
export const getReportDataSuccess = createAction(C.GET_REPORT_DATA_SUCCESS);
export const getReportDataFailure = createAction(C.GET_REPORT_DATA_FAILURE);

export const updateUsersRequest = createAction(C.UPDATE_USERS_REQUEST);
export const updateUsersSuccess = createAction(C.UPDATE_USERS_SUCCESS);
export const updateUsersFailure = createAction(C.UPDATE_USERS_FAILURE);
