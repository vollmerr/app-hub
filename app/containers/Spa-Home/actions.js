/*
 *
 * SpaHome actions
 *
 */

import {
  EXAMPLE_DATA_REQUEST,
  EXAMPLE_DATA_SUCCESS,
  EXAMPLE_DATA_FAILURE,
} from './constants';

export function exampleDataRequest() {
  return {
    type: EXAMPLE_DATA_REQUEST,
  };
}

export function exampleDataSuccess(data) {
  return {
    data,
    type: EXAMPLE_DATA_SUCCESS,
  };
}

export function exampleDataFailure(error) {
  return {
    error,
    type: EXAMPLE_DATA_FAILURE,
  };
}
