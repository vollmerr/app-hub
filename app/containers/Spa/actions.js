import { createAction } from 'redux-actions';

import {
  INIT_DATA_REQUEST,
  INIT_DATA_SUCCESS,
  INIT_DATA_FAILURE,
  NEW_ACK_REQUEST,
  NEW_ACK_SUCCESS,
  NEW_ACK_FAILURE,
  DISABLE_ACK_REQUEST,
  DISABLE_ACK_SUCCESS,
  DISABLE_ACK_FAILURE,
  GET_RECIPIENTS_REQUEST,
  GET_RECIPIENTS_SUCCESS,
  GET_RECIPIENTS_FAILURE,
} from './constants';
// INITAL DATA
export const initDataRequest = createAction(INIT_DATA_REQUEST);
export const initDataSuccess = createAction(INIT_DATA_SUCCESS);
export const initDataFailure = createAction(INIT_DATA_FAILURE);
// NEW ACKNOWLEDGMENT
export const newAckRequest = createAction(NEW_ACK_REQUEST);
export const newAckSuccess = createAction(NEW_ACK_SUCCESS);
export const newAckFailure = createAction(NEW_ACK_FAILURE);
// DSIABLE ACKNOWLEDGMENT
export const disableAckRequest = createAction(DISABLE_ACK_REQUEST);
export const disableAckSuccess = createAction(DISABLE_ACK_SUCCESS);
export const disableAckFailure = createAction(DISABLE_ACK_FAILURE);
// GET LIST OF RECIPIENTS FOR EXISTSING ACK
export const getRecipientsRequest = createAction(GET_RECIPIENTS_REQUEST);
export const getRecipientsSuccess = createAction(GET_RECIPIENTS_SUCCESS);
export const getRecipientsFailure = createAction(GET_RECIPIENTS_FAILURE);
