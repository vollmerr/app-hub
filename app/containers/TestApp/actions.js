import { createActions } from 'redux-actions';

import {
  INIT_DATA_REQUEST,
  INIT_DATA_SUCCESS,
  INIT_DATA_FAILURE,
  CLEAR_ERRORS,
} from './constants';

const actions = createActions({
  [INIT_DATA_REQUEST]: () => {},
  [INIT_DATA_SUCCESS]: (payload) => payload,
  [INIT_DATA_FAILURE]: (payload) => payload,
  [CLEAR_ERRORS]: () => {},
});

export default actions.app.testApp;
