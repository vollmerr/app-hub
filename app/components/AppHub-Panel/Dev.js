import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { authUser } from 'containers/AppHub/actions';
import { makeSelectUserSam } from 'containers/AppHub/selectors';
import theme from 'utils/theme';

import Wrapper from './Wrapper';


export const Field = styled.div`
  label {
    color: ${theme.white};
  }
`;


/**
 * Panel for development utilities, such as changing user jwt.
 * (This should be in containers, but makes more sense to have with other panels)
 */
export class Dev extends React.PureComponent {
  handleClearToken = () => {
    localStorage.removeItem('id_token');
    window.location.reload();
  }

  handleClickUser = (user) => {
    const { onAuthUser, onClick } = this.props;

    if (user && user.key) {
      localStorage.setItem('id_token', user.key);
    } else {
      localStorage.removeItem('id_token');
    }
    onAuthUser();
    onClick();
    window.location.reload();
  }

  render() {
    const { userSam } = this.props;
    const jwts = DEV.JWT; // eslint-disable-line

    return (
      jwts ?
        <Wrapper>
          <h2>Developer Options</h2>
          <hr />
          <DefaultButton
            text={'Clear Token'}
            onClick={this.handleClearToken}
          />
          <h3>Switch User</h3>
          <Field>
            <TextField
              disabled
              label={'Current User'}
              value={userSam}
            />
          </Field>
          {
            Object.keys(jwts).map((group) => (
              <Field key={group}>
                <Dropdown
                  label={`${group} Test Users`}
                  id={group}
                  options={jwts[group]}
                  onChanged={this.handleClickUser}
                />
              </Field>
            ))
          }
        </Wrapper> :
        null
    );
  }
}


const { func, string } = PropTypes;

Dev.propTypes = {
  onAuthUser: func.isRequired,
  userSam: string,
  onClick: func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  userSam: makeSelectUserSam(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onAuthUser: () => dispatch(authUser()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Dev);
