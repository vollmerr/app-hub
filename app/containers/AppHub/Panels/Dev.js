import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import { Dropdown } from 'office-ui-fabric-react/lib/Dropdown';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { compose } from 'redux';

import toJS from '../../../hocs/toJS';
import theme from '../../../utils/theme';
import * as api from '../../../utils/api';

import * as selectors from '../selectors';
import * as actions from '../actions';

import Wrapper from './Wrapper';


export const Field = styled.div`
  label {
    color: ${theme.white};
  }
`;


/**
 * Panel for development utilities, such as changing user jwt.
 */
export class Dev extends React.PureComponent {
  handleClearToken = () => {
    api.clearToken();
    window.location.reload();
  }

  handleClickUser = (user) => {
    const { onAuthUser, onClick } = this.props;

    if (user && user.key) {
      api.setToken(user.key);
    } else {
      api.clearToken();
    }
    onAuthUser();
    onClick();
    window.location.reload();
  }

  render() {
    const { user } = this.props;
    const jwts = DEV.JWT || MOCK.JWT; // eslint-disable-line

    return (
      jwts ?
        <Wrapper>
          <h2>Developer Options</h2>
          <hr />
          <DefaultButton
            text={'Clear Token'}
            onClick={this.handleClearToken}
          />
          <h3>Switch User</h3>
          <Field>
            <TextField
              disabled
              label={'Current User'}
              value={user.name}
            />
          </Field>
          <Field>
            <TextField
              disabled
              multiline
              rows={user.roles.length + 1}
              label={'Current Roles'}
              value={user.roles.join('\n')}
            />
          </Field>
          {
            Object.keys(jwts).map((group) => (
              <Field key={group}>
                <Dropdown
                  label={`${group} Test Users`}
                  id={group}
                  options={jwts[group]}
                  onChanged={this.handleClickUser}
                />
              </Field>
            ))
          }
        </Wrapper> :
        null
    );
  }
}


const { func, object } = PropTypes;

Dev.propTypes = {
  user: object.isRequired,
  onAuthUser: func.isRequired,
  onClick: func.isRequired,
};


const mapStateToProps = createStructuredSelector({
  user: selectors.getUser,
});

export function mapDispatchToProps(dispatch) {
  return {
    onAuthUser: () => dispatch(actions.authUser()),
  };
}

const withConnect = connect(mapStateToProps, mapDispatchToProps);


export default compose(
  withConnect,
  toJS,
)(Dev);
