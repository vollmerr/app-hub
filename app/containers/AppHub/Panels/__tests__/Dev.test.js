import React from 'react';
import { shallow } from 'enzyme';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

import { testStyledComponent } from '../../../../utils/testUtils';
import * as api from '../../../../utils/api';

import { authUser } from '../../actions';

import { Dev, Field, mapDispatchToProps } from '../Dev';


testStyledComponent(Field);


const user1 = { name: 'test name 1', key: 'test key 1' };
const user2 = { name: 'test name 3', key: 'test key 3' };

global.isDev = true;
global.isMock = true;
global.DEV = {
  JWT: {
    group1: [
      user1,
      { name: 'test name 2', key: 'test key 2' },
    ],
    group2: [
      user2,
      { name: 'test name 4', key: 'test key 4' },
    ],
  },
};
global.MOCK = {
  JWT: {
    group1: [
      user1,
      { name: 'test name 5', key: 'test key 5' },
    ],
    group2: [
      user2,
      { name: 'test name 7', key: 'test key 7' },
    ],
  },
};

const props = {
  user: {
    key: '123',
    name: 'user name',
    roles: ['role1', 'role2'],
  },
  onAuthUser: jest.fn(),
  onClick: jest.fn(),
};


describe('<Dev />', () => {
  let wrapper;
  let instance;
  beforeEach(() => {
    wrapper = shallow(<Dev {...props} />);
    instance = wrapper.instance();
  });

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not render if DEV.JWT/MOCK.JWT does not exist', () => {
      const jwt = { ...global.DEV.JWT };
      try {
        global.DEV.JWT = undefined;
        wrapper = shallow(<Dev {...props} />);
        expect(wrapper.equals(null)).toEqual(true);
      } finally {
        global.DEV.JWT = jwt;
      }
    });

    it('should use the `DEV.JWT` jwts if in dev mode', () => {
      expect(global.jwts).toEqual(global.DEV.JWT);
    });

    it('should use the `MOCK.JWT` jwts if in mock mode', () => {
      try {
        global.isDev = false;
        wrapper = shallow(<Dev {...props} />);
        expect(global.jwts).toEqual(global.MOCK.JWT);
      } finally {
        global.isDev = true;
      }
    });

    it('should not render the panel if in production (no global.isDev or global.isMock)', () => {
      try {
        global.isMock = false;
        global.isDev = false;
        wrapper = shallow(<Dev {...props} />);
        expect(global.jwts).toEqual(undefined);
      } finally {
        global.isMock = true;
        global.isDev = true;
      }
    });

    it('should render a h2 header', () => {
      expect(wrapper.find('h2').length).toEqual(1);
    });

    it('should render a button for clearing the token from localStorage', () => {
      expect(wrapper.find(DefaultButton).length).toEqual(1);
    });

    it('should render the groups in their own group', () => {
      const groups = wrapper.findWhere((x) => x.prop('options'));
      // there are 2 groups
      expect(groups.length).toEqual(2);
      // user 1 in group 1, but not user2
      expect(groups.at(0).prop('options').includes(user1)).toEqual(true);
      expect(groups.at(0).prop('options').includes(user2)).toEqual(false);
      // user 2 in group 2, but not user1
      expect(groups.at(1).prop('options').includes(user1)).toEqual(false);
      expect(groups.at(1).prop('options').includes(user2)).toEqual(true);
    });
  });

  describe('handleClearToken', () => {
    beforeEach(() => {
      api.clearToken = jest.fn();
      instance = wrapper.instance();
    });

    it('should exist', () => {
      expect(instance.handleClearToken).toBeDefined();
    });

    it('should remove the token from localstorage', () => {
      instance.handleClearToken();
      expect(api.clearToken).toHaveBeenCalled();
    });
  });

  describe('handleClickUser', () => {
    beforeEach(() => {
      api.setToken = jest.fn();
      api.clearToken = jest.fn();
      instance = wrapper.instance();
    });

    it('should exist', () => {
      expect(instance.handleClickUser).toBeDefined();
    });

    it('should dispatch onAuthUserRequest when called', () => {
      instance.handleClickUser(user1);
      expect(props.onAuthUser).toHaveBeenCalled();
    });

    it('should update the local token if a user with a key is passed', () => {
      instance.handleClickUser(user1);
      expect(api.setToken).toHaveBeenCalled();
    });

    it('should clear the local token if no user or user key', () => {
      instance.handleClickUser();
      expect(api.clearToken).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    let mappedDispatch;
    let dispatch;
    beforeEach(() => {
      dispatch = jest.fn();
      mappedDispatch = mapDispatchToProps(dispatch);
    });

    describe('onAuthUser', () => {
      it('should be injected', () => {
        expect(mappedDispatch.onAuthUser).toBeDefined();
      });

      it('should dispatch authUser when called', () => {
        mappedDispatch.onAuthUser();
        expect(dispatch).toHaveBeenCalledWith(authUser());
      });
    });
  });
});
