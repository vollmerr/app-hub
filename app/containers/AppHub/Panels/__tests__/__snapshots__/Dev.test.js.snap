// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dev /> render should render correctly 1`] = `
<Wrapper__Wrapper>
  <h2>
    Developer Options
  </h2>
  <hr />
  <CustomizedDefaultButton
    onClick={[Function]}
    text="Clear Token"
  />
  <h3>
    Switch User
  </h3>
  <Dev__Field>
    <TextField
      autoAdjustHeight={false}
      borderless={false}
      deferredValidationTime={200}
      disabled={true}
      errorMessage=""
      label="Current User"
      multiline={false}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      underlined={false}
      validateOnFocusIn={false}
      validateOnFocusOut={false}
      validateOnLoad={true}
      value="user name"
    />
  </Dev__Field>
  <Dev__Field>
    <TextField
      autoAdjustHeight={false}
      borderless={false}
      deferredValidationTime={200}
      disabled={true}
      errorMessage=""
      label="Current Roles"
      multiline={true}
      onBeforeChange={[Function]}
      onChanged={[Function]}
      onGetErrorMessage={[Function]}
      onNotifyValidationResult={[Function]}
      resizable={true}
      rows={3}
      underlined={false}
      validateOnFocusIn={false}
      validateOnFocusOut={false}
      validateOnLoad={true}
      value="role1
role2"
    />
  </Dev__Field>
  <Dev__Field
    key="group1"
  >
    <WithResponsiveMode
      id="group1"
      label="group1 Test Users"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "test key 1",
            "name": "test name 1",
          },
          Object {
            "key": "test key 2",
            "name": "test name 2",
          },
        ]
      }
    />
  </Dev__Field>
  <Dev__Field
    key="group2"
  >
    <WithResponsiveMode
      id="group2"
      label="group2 Test Users"
      onChanged={[Function]}
      options={
        Array [
          Object {
            "key": "test key 3",
            "name": "test name 3",
          },
          Object {
            "key": "test key 4",
            "name": "test name 4",
          },
        ]
      }
    />
  </Dev__Field>
</Wrapper__Wrapper>
`;

exports[`<Dev__Field /> should render correctly 1`] = `
.c0 label {
  color: #ffffff;
}

<div
  className="c0"
/>
`;
